packages/storage/src/providers/s3/utils/getUrlDurationInSeconds.ts

```diff
@@ -0,0 +1,13 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+export function getUrlDurationInSeconds(
+	awsCredExpiration: Date,
+	urlExpiration: number
+): number {
+	urlExpiration =
+		awsCredExpiration.getTime() / 1000 < urlExpiration
```

Comment:
Not exact code but you need to calculate the the time to expire to compare it with `urlExpiration`. 
```suggestion
		(awsCredExpiration.getTime() - Date.now()) / 1000 < urlExpiration
```

Reply:
Fixed Thanks for the clarification